app.js

```diff
@@ -0,0 +1,67 @@
+const express = require( 'express' )
+const path = require( 'path' )
+const favicon = require( 'serve-favicon' )
+const logger = require( 'morgan' )
+const cookieParser = require( 'cookie-parser' )
+const bodyParser = require( 'body-parser' )
+const db = require( './database/database' )
+
+const routes = require( './routes/index' )
+const users = require( './routes/users' )
+const apiRoutes = require( './routes/api/v1/users' )
+
+const appRoot = process.env.APP_ROOT
+const app = express()
+
+app.set( 'views', path.join( __dirname, './views' ))
+app.set( 'view engine', 'pug' )
+
+app.set( 'env', process.env.PORT || '3000' )
+if( process.env.NODE_ENV !== 'test' ) app.use( logger( 'dev' ))
+
+app.use( bodyParser.json() )
+
+app.use( logger( 'dev' ))
+app.use( bodyParser.json() )
+app.use( bodyParser.urlencoded({ extended: false }))
+app.use( cookieParser())
+
+app.use( express.static( path.join( __dirname, './front_end/public' )))
+
+app.use( '/', routes )
+app.use( '/users', users )
+app.use( '/api/v1/users', apiRoutes )
+
+app.use(( request, response, next ) => {
```

Comment:
Let's define this as middleware in another file to help keep this file clean:
```js
const cors = ( request, response, next ) => {
  // all the things
  next()
}
```
Then we can import this as `cors` in `app.js` and:
```js
const cors = require( './src/cors' ) // or whatever
app.use( cors )
```